tools/src/main/scala/scala/scalanative/linker/Reach.scala

```diff
@@ -849,12 +849,14 @@ class Reach(
   }
 
   protected def addMissing(global: Global, pos: Position): Unit = {
-    val prev = missing.getOrElse(global, Set.empty)
-    val position = NonReachablePosition(
-      path = Paths.get(pos.source),
-      line = pos.line + 1
-    )
-    missing(global) = prev + position
+    val prev = missing.getOrElseUpdate(global, Set.empty)
+    if (pos != nir.Position.NoPosition) {
```

Comment:
Isn't this going to *ignore* missing stuff references from an empty position altogether, potentially letting the linker phase to complete without error where there should be one? If that's the case, we should instead at least put something in `missing(global)`, even if it is with a `path: None` or something like that.

Reply:
It would not, `missing(global)` is set using `missing.getOrElseUpdate(global, Set.empty)`. It means we know it is missing but don't know where, because it might happen that for some reason the position is empty, eg. when modified in the scalac phase without setting a new position

Reply:
Ah OK. Looks good, then.